src/main/java/net/minecraftforge/fmp/microblock/IMicroblockPlacementGrid.java

```diff
@@ -0,0 +1,21 @@
+package net.minecraftforge.fmp.microblock;
+
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+/**
+ * Represents a grid for {@link IMicroblock} placement.
+ *
+ * @see IMicroblock
+ * @see MicroblockClass
+ */
+@SideOnly(Side.CLIENT)
```

Comment:
It's the grid that's rendered on the ground when attempting to place a microblock.
Most of the microblock specification is there for compatibility purposes, even though it's not really required. But this way all mods have a common way to interact with any others that may add their own microblocks.